File: cpp/Dynamic Programming/L_coinChange2.cpp
#include <bits/stdc++.h>
using namespace std;

int change(int amount, vector<int> &coins)
{
    vector<int> changeMethodCount(amount + 1, 0);
    changeMethodCount[0] = 1;
    for (int coin : coins)
    {
        for (int smallAmount = coin; smallAmount <= amount; smallAmount++)
        {
            changeMethodCount[smallAmount] += changeMethodCount[smallAmount - coin];
        }
    }
    return changeMethodCount[amount];
}

File: Python/Dynamic Programming/L_coinChange2.py
from typing import List


def change(amount: int, coins: List[int]) -> int:
    change_method_count = [1] + [ 0 for _ in range(amount)]
    for cur_coin in coins:
        for small_amount in range(cur_coin, amount+1):
            change_method_count[small_amount] += change_method_count[small_amount - cur_coin]
    return change_method_count[amount]